portfolio/src/main/webapp/script.js

```diff
@@ -55,6 +56,46 @@ function showFrame() {
 }
 
 function getComments() {
+  myFetch();
+}
+
+async function myFetch() {
+  const comments = await fetch('/data').then((response) => response.json());
+  const commentsList = document.getElementById('comments-container');
+  commentsList.innerHTML = '';
+  const maxInput = document.getElementById('max').value;
+  const paramInput = 'max='.concat(maxInput);
+  const params = new URLSearchParams(paramInput);
+  let max = params.get('max');
+  if (max == 'all' || max > comments.length) {
+    max = comments.length;
+  }
+  const fetches = [];
+  for (let i = 0; i < max; i++) {
+    if (comments[i].hasOwnProperty('blobKey') &&
+        comments[i].blobKey !== 'noBlob') {
+      fetches.push(
+          fetch('/getBlob?blobKey=' + comments[i].blobKey).then((imgBlob) => {
+            comments[i].url = imgBlob.url;
+          }));
+    }
+  }
+  Promise.all(fetches).then(() => {
+    for (let i = 0; i < max; i++) {
+      console.log(comments[i]);
+      commentsList.append(createCommentElement(comments[i]));
+    }
+  });
+}
+
+function showLogin() {
```

Comment:
Why do you have both `getLogin()` and `showLogin()`? Can you use just one?

Reply:
So getLogin returns the status of whether or not someone is logged in, and it throws an error when someone isn't. Therefore, it doesn't get to the end of the code (where I could put the showLogin stuff). I tried doing it with a try catch function, but it ended up looking more messy than having the two separate functions... Is it ok if I keep them?